src/Operations.jl

```diff
@@ -900,8 +900,9 @@ function update_package_add(pkg::PackageSpec, entry::PackageEntry, is_dep::Bool)
     if entry.path !== nothing || entry.repo.url !== nothing || pkg.repo.url !== nothing
         return pkg # overwrite everything, nothing to copy over
     end
-    if is_dep && ((isa(pkg.version, VersionNumber) && entry.version == pkg.version) ||
-                 (!isa(pkg.version, VersionNumber) && entry.version ∈ pkg.version))
+    if is_dep && !is_stdlib(pkg.uuid) &&
```

Comment:
I would prefer if the stdlib branch was separate. E.g. `is_stdlib(pkg.uuid) && return pkg`

Reply:
Otherwise looks good

Reply:
Yea, that is cleaner.